gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+        return ehrExtract
+            .getComponent()
+            .stream()
+            .filter(EhrResourceExtractorUtil::hasEhrFolder)
+            .map(RCMRMT030101UK04Component::getEhrFolder)
+            .map(RCMRMT030101UK04EhrFolder::getComponent)
+            .flatMap(List::stream)
+            .filter(EhrResourceExtractorUtil::hasEhrComposition)
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .filter(ehrComposition -> isEncounterEhrComposition(ehrComposition.getCode().getCode(), ehrComposition.getComponent()))
+            .toList();
+    }
+
+    private boolean isEncounterEhrComposition(String code, List<RCMRMT030101UK04Component4> component) {
+        return !INVALID_CODES.contains(code) && !component.stream().anyMatch(this::hasSuppressedContent);
+    }
+
+    private boolean hasSuppressedContent(RCMRMT030101UK04Component4 component) {
+        return component.getEhrEmpty() != null || component.getRegistrationStatement() != null;
+    }
+
+    private List<RCMRMT030101UK04CompoundStatement> getTopicCompoundStatements(RCMRMT030101UK04EhrComposition ehrComposition) {
+        return ehrComposition
+            .getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component4::getCompoundStatement)
+            .filter(compoundStatement -> compoundStatement != null && compoundStatement.getClassCode().get(0).equals(TOPIC_CLASS_CODE))
+            .toList();
+    }
+
+    private Encounter mapToEncounter(RCMRMT030101UK04EhrComposition ehrComposition, Patient patient) {
+        var id = ehrComposition.getId().getRoot();
+
+        var encounter = new Encounter();
+        encounter
+            .setParticipant(getEncounterParticipant(ehrComposition.getAuthor(), ehrComposition.getParticipant2()))
+            .setStatus(EncounterStatus.FINISHED)
+            .setSubject(new Reference(patient))
+            .setType(getEncounterType(ehrComposition.getCode()))
+            .setPeriod(getPeriod(ehrComposition.getEffectiveTime(), ehrComposition.getAvailabilityTime()))
+            .setIdentifier(getEncounterIdentifier(id))
+            .setMeta(getEncounterMeta())
+            .setId(id);
+
+        setEncounterLocation(encounter, ehrComposition);
+
+        return encounter;
+    }
+
+    private List<CodeableConcept> getEncounterType(CD code) {
+        return List.of(codeableConceptMapper.mapToCodeableConcept(code));
+    }
+
+    private Meta getEncounterMeta() {
+        return new Meta().addProfile(ENCOUNTER_META_PROFILE);
+    }
+
+    private List<Identifier> getEncounterIdentifier(String id) {
+        Identifier identifier = new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+        return List.of(identifier);
+    }
+
+    private Period getPeriod(IVLTS effectiveTime, TS availabilityTime) {
```

Comment:
Could you not pass in the ehrComposition here? it appears to be the only values you pass in